drivers/auxiliary/ipx800.h

```diff
@@ -0,0 +1,224 @@
+	
+    virtual bool UpdateDigitalInputs() override;
+    virtual bool UpdateAnalogInputs() override; // IPX800 Analog Inputs not managed
+    virtual bool UpdateDigitalOutputs() override;
+    virtual bool CommandOutput(uint32_t index, OutputState command) override;
+	virtual bool saveConfigItems(FILE *fp) override;
+	
+  private:
+	bool roofPowerManagement = false;
+	Connection::TCP *tcpConnection {nullptr};
+	
+	// TO manage Password in a next release
+	//IText* getMyLogin();
+    //ITextVectorProperty getMyLoginVector();
+    
+	ISwitchVectorProperty getMyRelayVector(int i);
+    ISwitchVectorProperty getMyDigitsVector(int i);
+	
+	// List of possible commands 
+    enum {
+        UNUSED_RELAY,
+        ROOF_ENGINE_POWER_SUPPLY,
+        TUBE_VENTILATION,
+        HEATING_RESISTOR_1,
+        HEATING_RESISTOR_2,
+        ROOF_CONTROL_COMMAND,
+        MOUNT_POWER_SUPPLY,
+        CAM_POWER_SUPPLY,
+        OTHER_POWER_SUPPLY_1,
+        OTHER_POWER_SUPPLY_2,
+        OTHER_POWER_SUPPLY_3
+        } IPXRelaysCommands;
+
+	// List of possibles digital inputs
+    enum  {
+        UNUSED_DIGIT,
+        DEC_AXIS_PARKED,
+        RA_AXIS_PARKED,
+        ROOF_OPENED,
+        ROOF_CLOSED,
+        ROOF_ENGINE_POWERED,
+        RASPBERRY_SUPPLIED,
+        MAIN_PC_SUPPLIED,
+        OTHER_DIGITAL_1,
+        OTHER_DIGITAL_2 } IPXDigitalRead;
+	
+    char tmpAnswer[8]= {0};
+    bool setupParams();
+    float CalcTimeLeft(timeval);
+
+    ISState fullOpenLimitSwitch { ISS_ON };
+    ISState fullClosedLimitSwitch { ISS_OFF };
+    double MotionRequest { 0 };
+    struct timeval MotionStart { 0, 0 };
+
+    ISwitch RelaisInfoS[11] {};
+    ISwitch Relais1InfoS[11], Relais2InfoS[11],Relais3InfoS[11],Relais4InfoS[11],Relais5InfoS[11],Relais6InfoS[11],Relais7InfoS[11],Relais8InfoS[11] {};
+    ISwitchVectorProperty RelaisInfoSP[8] {};
+
+    ISwitch DigitalInputS[10] {};
+    ISwitch Digital1InputS[10], Digital2InputS[10], Digital3InputS[10], Digital4InputS[10], Digital5InputS[10], Digital6InputS[10], Digital7InputS[10], Digital8InputS[10];
+    ISwitchVectorProperty DigitalInputSP[8] {};
+
+    ISwitch Relay1StateS[2],Relay2StateS[2],Relay3StateS[2],Relay4StateS[2],Relay5StateS[2],Relay6StateS[2],Relay7StateS[2],Relay8StateS[2] ;
+    ISwitchVectorProperty RelaysStatesSP[8] ;
+    ISwitch Digit1StateS[2],Digit2StateS[2],Digit3StateS[2],Digit4StateS[2],Digit5StateS[2],Digit6StateS[2],Digit7StateS[2],Digit8StateS[2] ;
+    ISwitchVectorProperty DigitsStatesSP[8] ;
+
+    //TO manage Password in a next release
+    //IText LoginPwdT[2];
+    //ITextVectorProperty LoginPwdTP;
+	//std::string myPasswd = "";
+    //std::string myLogin = "";
+
+    enum {
+        ROOF_IS_OPENED ,
+        ROOF_IS_CLOSED ,
+        UNKNOWN_STATUS
+    }
+    Roof_Status;
+
+    enum {
+        RA_PARKED    ,
+        DEC_PARKED   ,
+        BOTH_PARKED  ,
+        NONE_PARKED
+    }
+    Mount_Status;
+
+	const char *ROLLOFF_TAB        = "Roll Off";
```

Comment:
There shouldn't be anything related to roof or dome in the driver, it's a generic input/output driver.

Reply:
cf. previous comment.